### PR TITLE
feat: upgrade client npm packages

### LINKED ISSUE #14077: [Task]:  Update Dependencies
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Update Dependencies showing vulnerabilities in the Dependentbot alerts.
- Merge Pull Requests from Dependent Bot

### PR DESCRIPTION
**Pull request by yaldram**
## Description

- Update Dependencies showing vulnerabilities in dependent-bot alerts.

Fixes #14077 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

- Tested by running the app using `yarn start`
- Tested by building the app locally.
- Ran some cypress tests locally.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/.eslintrc.json [hunk 1]
```diff
@@ -26,10 +26,10 @@
     "import/no-webpack-loader-syntax": 0,
     "no-undef": 0,
     "react/prop-types": 0,
+    "react/display-name": 0,
     "@typescript-eslint/explicit-module-boundary-types": 0,
     "cypress/no-unnecessary-waiting": 0,
     "cypress/no-assigning-return-values": 0,
-    "react/function-component-definition": "warn",
```
Comment (riodeuno) [inline app/client/.eslintrc.json:32]: Is there a reason we have removed this?

File: app/client/craco.common.config.js [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+        rules: [
+          {
+            test: /\.m?js/,
+            resolve: { fullySpecified: false },
+          },
+        ],
+      },
+      ignoreWarnings: [
+        function ignoreSourcemapsloaderWarnings(warning) {
+          return (
```
Comment (riodeuno) [inline app/client/craco.common.config.js:33]: What kind of warnings were you facing, which lead to adding this block of code?

Comment (yaldram) [reply app/client/craco.common.config.js:33]: Yes these warnings were for the sourcemaps for the source map loader - still an open PR - https://github.com/facebook/create-react-app/pull/11752

**Comment by riodeuno** [comment]
@yaldram I've converted this to draft and removed reviewers for now. Once we have the CI working on this branch, we can re-open this.

File: app/client/.eslintrc.json (hunk 1 above)
Comment (yaldram) [reply app/client/.eslintrc.json:32]: This rule was conflicting with the eslint version for me, it was giving warnings though the code was right.

**Comment by yaldram** [comment]
Closed in favor of - https://github.com/appsmithorg/appsmith/pull/14858

**Pull request closed without merging**